biopet-core/src/main/scala/nl/lumc/sasc/biopet/core/PedigreeQscript.scala

```diff
@@ -0,0 +1,181 @@
+package nl.lumc.sasc.biopet.core
+
+import java.io.PrintWriter
+
+import nl.lumc.sasc.biopet.core.PedigreeQscript.PedMergeStrategy
+import org.broadinstitute.gatk.queue.QScript
+
+import scala.io.Source
+
+/**
+ * Created by Sander Bollen on 28-3-17.
+ */
+trait PedigreeQscript extends MultiSampleQScript { qscript: QScript =>
+
+  /* Optionally parse from ped file */
+  def ped: Option[File] = None
+
+  /* The merge stategy to use when we have both a ped file and sample tag information */
+  def mergeStrategy: PedMergeStrategy.Value = PedMergeStrategy.Concatenate
+
+  /**
+   * Case class representing a PED samples
+   * For the PED format, see:
+   * http://pngu.mgh.harvard.edu/~purcell/plink/data.shtml
+   * @param familyId family id
+   * @param individualId individual id
+   * @param paternalId Optional paternal id
+   * @param maternalId Optional maternal id
+   * @param gender gender
+   * @param affectedPhenotype Optional boolean
+   * @param genotypeFields optional genotype fileds
+   */
+  case class PedSample(familyId: String, individualId: String,
+                       paternalId: Option[String],
+                       maternalId: Option[String],
+                       gender: MultiSampleQScript.Gender.Value,
+                       affectedPhenotype: Option[Boolean],
+                       genotypeFields: List[String] = Nil)
+
+  lazy val pedSamples: List[PedSample] = {
+    mergeStrategy match {
+      case PedMergeStrategy.Concatenate => pedSamplesFromConfig() ::: parsePedFile()
+      case _                            => throw new NotImplementedError() // todo
+    }
+  }
+
+  /**
+   * Get affected samples
+   *
+   * @return List[PedSample]
+   */
+  def getIndexSamples: List[PedSample] = {
+    pedSamples.filter(x => x.affectedPhenotype)
+  }
+
+  /**
+   * Get pedSamples from sample tags in config
+   * May return empty list if no pedigree can be constructed
+   *
+   * @return
+   */
+  def pedSamplesFromConfig(): List[PedSample] = {
+    val totalSampleIds = samples.values.map(_.sampleId).toList
+    samples.values.filter(x => x.father.isDefined && x.mother.isDefined && x.family.isDefined).map { x =>
```

Comment:
Does this mean that if any of the samples do not have a father, a mother, and a family, it will be silently ignored?